src/common/data-modeler-state-service/TableStateActions.ts

```diff
@@ -11,10 +11,7 @@ export class TableStateActions extends StateActions {
     @StateActions.PersistentTableAction()
     public addOrUpdateTableToState({stateService, draftState}: PersistentTableStateActionArg,
                                    table: PersistentTableEntity, isNew: boolean): void {
-        if (isNew) {
-            stateService.addEntity(draftState, table);
-        } else {
-            stateService.updateEntity(draftState, table.id, table);
-        }
+        // we do not create any state here since it is contingent on whether the import step
```

Comment:
@AdityaHegde let me know what you think about this. The issue with how we're utilizing state actions is they can be pretty inflexible. We can't create or update the table state until _after_ we know for sure the table has imported successfully. Yet we also tie the import status to an existing table entity. The mental model here does not grant us the flexibility we need to do the things we need to do. Instead I create or update the table in `TableActions.ts` and then revert the change from within the action if the table import fails. While this is more code, it's comprehensible what's happening.

The main downside is that we are still even in this case too rigid. The `lastUpdated` field is updated somewhere else in the code, with no way for me to specify that I want to revert the table update fully.  The table entity was _not_ in fact last updated when the timestamp says.